src/WaveTrack.cpp

```diff
@@ -1389,8 +1389,12 @@ void WaveTrack::SyncLockAdjust(double oldT1, double newT1)
       {
          // Check if clips can move
          if (EditClipsCanMove.Read()) {
-            auto tmp = Cut (oldT1, GetEndTime() + 1.0/GetRate());
-            Paste(newT1, tmp.get());
+            const auto offset = newT1 - oldT1;
+            for(const auto& clip : mClips)
+            {
+               if (clip->GetPlayStartTime() > oldT1 - (1.0 / mRate))
```

Comment:
What is the reason for the correction by a sample duration?
Might it be better to substract one-half of that amount?

Reply:
Yes, correct, that should be one-half
Thanks!

Reply:
Or, not... it is possible that there could be 1 sample-length clip that has a beginning at `oldT1 - 1/rate`, and at such clip should not be moved (this is guaranteed by `>`), but considering that clips aren't bound to be aligned to a track's sample grid, it is actually possible that some clip starts somewhere between `oldT1 - 1/rate` and `oldT1 - 0.5/rate` and such clip should be adjusted